lib/assert.js

```diff
@@ -222,13 +223,25 @@ function wrapAssertions(callbacks) {
 				pending(this, intermediate);
 				// Don't reject the returned promise, even if the assertion fails.
 				return intermediate.catch(noop);
+			};
+
+			if (promise) {
+				return handlePromise.call(this, promise);
 			}
 
 			try {
 				const retval = test(fn);
 				pass(this);
 				return retval;
 			} catch (err) {
+				if (maybePromise) {
+					if (isPromise(maybePromise)) {
+						return handlePromise.call(this, maybePromise);
+					}
+					if (isObservable(maybePromise)) {
+						return handlePromise.call(this, observableToPromise(maybePromise));
```

Comment:
Could you add a test for this code path?

Reply:
sure, we can use already provided library `zen-observable` in package.json